utils/sample_qc.py

```diff
@@ -364,82 +364,139 @@ def infer_sex(
 
 
 def get_sex_expr(
+    ht: hl.Table,
     chr_x_ploidy: hl.expr.NumericExpression,
     chr_y_ploidy: hl.expr.NumericExpression,
     f_stat: hl.expr.NumericExpression,
-    xx_x_ploidy_cutoffs: Tuple[float, float] = (1.4, 2.25),
-    xy_x_ploidy_cutoffs: Tuple[float, float] = (0.5, 1.4),
-    xx_y_ploidy_cutoff: float = 0.1,
-    xy_y_ploidy_cutoffs: Tuple[float, float] = (0.15, 1.2),
-    yy_y_ploidy_cutoff: float = 1.3,
-    xxx_x_ploidy_cutoff:float = 2.5,
-    f_stat_female_cutoff: float = -0.2,
-    f_stat_male_cutoff: float = 0.2
+    f_stat_XX_cutoff: float = -0.2,
+    f_stat_XY_cutoff: float = 0.2,
+    lower_SD_cutoff: int = 5,
+    upper_SD_cutoff: int = 6,
+    Y_lower_bound: int = 2,
 ) -> hl.expr.StructExpression:
-    # TODO: Automate cutoffs
-    # TODO: Provide alternatives in case of missing annotations (e.g. no chr_y_ploidy)
     """
     Creates a struct with the following annotations:
-    - karyoptype (str)
-    - sex (str), which can be either 'male', 'female' or missing
+    - sex_karyoptype (str)
+    - sex (str), which can be either 'male', 'female', 'ambiguous_sex', or 'sex_aneuploidy'
     - is_female (missing for karyotypes that aren't either 'XX' or 'XY')
 
+    :param Table ht: Table with samples, their f-stats, and their sex chromosome ploidies
     :param NumericExpression chr_x_ploidy: chrom X ploidy (or relative ploidy)
     :param NumericExpression  chr_y_ploidy: chrom Y ploidy (or relative ploidy)
-    :param NumericExpression f_stat: chrom X F-stat
-    :param Tuple[float, float] xx_x_ploidy_cutoffs: Boundaries around the chom X ploidy for females
-    :param Tuple[float, float] xy_x_ploidy_cutoffs: Boundaries around the chom X ploidy for males
-    :param float xx_y_ploidy_cutoff: Max y ploidy for females
-    :param Tuple[float, float] xy_y_ploidy_cutoffs: Boundaries around the chom Y ploidy for males
-    :param float yy_y_ploidy_cutoff: Min chrom Y ploidy for YY
-    :param float xxx_x_ploidy_cutoff: Min chrom X ploidy for XXX
-    :param float f_stat_female_cutoff: Min f-stat for females
-    :param flat f_stat_male_cutoff: Max f-stat for males
-    :return: Struct expression with sex anotations
+    :param NumericExpression f_stat: chrom X f-stat
+    :param float f_stat_XX_cutoff: Min f-stat for XX samples
+    :param flat f_stat_XY_cutoff: Max f-stat for XY samples
+    :param int lower_SD_cutoff: Number of standard deviations to use when determining upper limit of sex chromosome ploidy cutoffs
+    :param int upper_SD_cutoff: Permissive number of standard deviations to use when determining lower ploidy limit and aneuploidy cutoffs
+    :param int Y_lower_bound: Number of standard deviations to use as the lower bound for Y, specifically to correct for somatic loss of Y
     :rtype: StructExpression
     """
+    def get_ploidy_cutoffs(ht: hl.Table, sex: str, f_stat_cutoff: float, lower_SD_cutoff: int, upper_SD_cutoff: int, Y_lower_bound: int) -> dict:
```

Comment:
I agree the field names might not exist in the ht, but I think I need to pass the ht in for this function; maybe it would be easiest to sit down and talk about this?

Reply:
I added a note about field name assumptions for the ht. As the code is now, `impute_sex_ploidy` should return `chrX_ploidy` and `chrY_ploidy`